src/block.rs

```diff
@@ -552,61 +552,93 @@ impl Block {
 }
 
 #[derive(Copy, Clone, Debug, PartialEq)]
-pub enum Comparator {
+pub enum Operator {
+    /// Can perhaps be replaced by wrapping the operator in an option.
     None,
-    /// Eq is also Assign
+    /// The = operator. Can be used as assignment, or to open a scope.
```

Comment:
This is what I mean by impedance mismatch. This comment should say that Eq is the equality operator. And calling the one for == "Equals" is just confusing because it's so rarely used and calling it "Equals" makes it step on the domain of Eq.

I don't think these new names are an improvement. `MaybeEq` and `Equals` are misleading, which is worse than cryptic.

Reply:
" This comment should say that Eq is the equality operator. " But it isn't the equality operator. That enum value doesn't represent that. Unless you want to make it represent both equality and assignment. In that case, we might as well get right of the other one, the one currently called Equals. There is no point in having two enum values that have overlapping meanings.

As for MaybeEq, why is that misleading? I couldn't think of a different name. Is there an official name for that operator?

Reply:
I renamed it `ConditionalEquals` for now. Maybe it could be called `QEquals` instead? While slightly more cryptic, it is a lot shorter.